src/Providers/YFrameServiceProviders.php

```diff
@@ -64,4 +66,33 @@ private function resolveCommands()
             ]);
         }
     }
+
+    protected function registerDirectives()
+    {
+        Blade::directive('yframe', function ($expression) {
+            list($url, $options) = $this->extractArguments($expression);
+
+            $width = $this->printMethodChain('width', $options);
+            $height = $this->printMethodChain('height', $options);
+            $isFullscreen = $this->printMethodChain('isFullscreen', $options);
+
+            return sprintf('<?php echo (new \SOS\LaravelYoutubeFrameGenerator\Classes\YFrame())%s%s%s->generate(%s); ?>', $width, $height, $isFullscreen, $url);
+        });
+    }
+
+    private function printMethodChain($key, $array)
+    {
+        return array_key_exists($key, $array) ? "->$key($array[$key])" : '';
+    }
+
+    private function extractArguments($arguments)
+    {
+        list($url, $options) = explode(',', $arguments, 2);
+        $options = collect(explode(',', trim(trim(str_replace(['[', ']'], ['', ''], $options)), ','))) // remove array brackets, trim trailing comma
```

Comment:
please add the comments before each declaration, this help us keep the style ci clean

Reply:
also here
![image](https://user-images.githubusercontent.com/37983260/152598132-18d4d2f7-20cc-453c-8e86-7010c2498cb9.png)

Reply:
Not actually, I was mean these comments before the options variable
![Screenshot 2022-02-04 222901](https://user-images.githubusercontent.com/71431942/152598793-007b98d2-89f8-4ab6-9155-6dc455bc9bcb.png)